drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+if __name__ == '__main__':
+    import sys
+    import rospy
+    import rospkg
+    import rosparam
+    from sub8_thruster_comm import thruster_comm_factory
+    from mil_misc_tools.text_effects import *
+    from mil_misc_tools.terminal_input import get_ch
+    
+    rospack = rospkg.RosPack()
+    
+    def ports_from_layout(layout):
+            '''Load and handle the thruster bus layout'''
+            port_dict = {}
+    
+            for port in layout['thruster_ports']:
+                thruster_port = thruster_comm_factory(port, layout['thrusters'], fake=True)
+    
+                # Add the thrusters to the thruster dict
+                for thruster_name, thruster_info in port['thrusters'].items():
+                    port_dict[thruster_name] = thruster_port
+    
+            return port_dict
+
+    rospy.init_node('thruster_spin_test')
+
+    sub8_thruster_mapper = rospack.get_path('sub8_thruster_mapper')
+    thruster_layout = rosparam.load_file(sub8_thruster_mapper + '/config/thruster_layout.yaml')[0][0]
+    thruster_ports = ports_from_layout(thruster_layout)
+    print len(thruster_ports)
+    print thruster_ports
+    names_from_motor_id = {0: 'FLH', 1: 'FLV', 2: 'FRH', 3: 'FRV',
+                           4: 'BLH', 5: 'BLV', 6: 'BRH', 7: 'BRV'}
+    active_thrusters = set()
+
+    fprint = FprintFactory(title='thruster_spin_test', time=rospy.Time.now, auto_bold=False).fprint
+    usage_msg = \
+    '''
+    Welcome to the thruster_spin_test tool.
+    Instructions:
+    * press up or down to control the thrust commanded
+    * press the thruster id to spin that specific thruster (1-indexed, 0 means all)
+    * if a thruster is already spinning, press its id again to stop it
+    '''
+    fprint(usage_msg)
+
+    thrust = 0.5  # Normalized thrust in [0, 1]
+    key = None
+
+    def command_thrusters(timer_event):
+        for motor_id in active_thrusters:
+            name = names_from_motor_id[motor_id]
+            thruster_ports[name].send_thrust_msg(motor_id, thrust)
```

Comment:
I did it down there so that I could display the new thrust value whenever it is changed.